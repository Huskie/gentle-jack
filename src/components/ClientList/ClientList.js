import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';
import { withTheme } from 'emotion-theming';

import { Container } from '../';

const StyledWrapper = styled('div')`
    background: #06f;
    color: ${props => props.theme.colours.delta};
    box-sizing: border-box;
    display: none;
    padding: 2rem 0;
    h2 {
        font-size: 1.125rem;
        letter-spacing: 1px;
        text-align: center;
        text-transform: uppercase;
    }
`;

const StyledList = styled('div')`
    display: flex;
    span {
        flex: 1;
    }
`;

const BaseClientList = ({ clients, title }) => (
    <StyledWrapper>
        <Container>
            {title && <h2>{title}</h2>}
            <StyledList>
                {clients.map(({ name }) => <span key={name}>{name}</span>)}
            </StyledList>
        </Container>
    </StyledWrapper>
);

BaseClientList.propTypes = {
    clients: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string
    })).isRequired,
    title: PropTypes.string
};

BaseClientList.defaultProps = {};

const ClientList = withTheme(BaseClientList);

export { ClientList };