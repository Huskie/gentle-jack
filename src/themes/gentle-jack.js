const theme = {
  borderRadius: '4px',
  colours: {
    alpha: '#2742b8',
    beta: '#222',
    gamma: '#06f',
    delta: '#fff',
    epsilon: '#1b6cff',
    zeta: '#ffd561',
    eta: '#4048ef',
  },
  gradients: {
    alpha: '#4048ef, #2742b8',
  },
  spacing: {
    componentMargin: '16px',
  },
  sizing: {
    maxWidth: '1080px',
  },
};

export { theme };
