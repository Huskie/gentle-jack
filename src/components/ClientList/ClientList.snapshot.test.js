import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'emotion-theming';

import { ClientList } from './ClientList';
import { theme } from '../../themes/gentle-jack';

const clients = [
  { name: 'Client A' },
  { name: 'Client B' },
  { name: 'Client C' },
  { name: 'Client X' },
  { name: 'Client Y' },
  { name: 'Client Z' },
];

describe('ClientList snapshots:', () => {
  it('component renders with defaults', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <ClientList clients={clients} />
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('component renders with all props set', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <ClientList clients={clients} title="Test title" />
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
