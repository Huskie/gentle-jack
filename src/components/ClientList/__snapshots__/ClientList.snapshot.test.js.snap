// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientList snapshots: component renders with all props set 1`] = `
.emotion-2 {
  background: #06f;
  color: #fff;
  box-sizing: border-box;
  display: none;
  padding: 2rem 0;
}

.emotion-2 h2 {
  font-size: 1.125rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  text-align: center;
  text-transform: uppercase;
}

.emotion-1 {
  margin: 0 auto;
  max-width: 1080px;
  padding: 0 24px;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 span {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-1"
  >
    <h2>
      Test title
    </h2>
    <div
      className="emotion-0"
    >
      <span>
        Client A
      </span>
      <span>
        Client B
      </span>
      <span>
        Client C
      </span>
      <span>
        Client X
      </span>
      <span>
        Client Y
      </span>
      <span>
        Client Z
      </span>
    </div>
  </div>
</div>
`;

exports[`ClientList snapshots: component renders with defaults 1`] = `
.emotion-2 {
  background: #06f;
  color: #fff;
  box-sizing: border-box;
  display: none;
  padding: 2rem 0;
}

.emotion-2 h2 {
  font-size: 1.125rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  text-align: center;
  text-transform: uppercase;
}

.emotion-1 {
  margin: 0 auto;
  max-width: 1080px;
  padding: 0 24px;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 span {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-1"
  >
    
    <div
      className="emotion-0"
    >
      <span>
        Client A
      </span>
      <span>
        Client B
      </span>
      <span>
        Client C
      </span>
      <span>
        Client X
      </span>
      <span>
        Client Y
      </span>
      <span>
        Client Z
      </span>
    </div>
  </div>
</div>
`;
